app/src/main/java/com/duckduckgo/app/browser/BrowserViewModel.kt

```diff
@@ -361,19 +367,21 @@ class BrowserViewModel @Inject constructor(
     }
 
     fun onSystemDefaultBrowserDialogSuccess() {
-        defaultBrowserPromptsExperiment.onSystemDefaultBrowserDialogSuccess()
+        defaultBrowserPromptsExperiment.onSystemDefaultBrowserDialogSuccess(
+            lastSystemDefaultBrowserDialogTrigger ?: SetAsDefaultActionTrigger.UNKNOWN,
```

Comment:
In practice, we shouldn't need to fall back to `UNKNOWN`. I added it as a safeguard because after briefly looking at the problem I couldn't find a non-nullable way to round-trip a parameter through the "activity for result" stack.

Since we can't launch the default browser dialog or system apps and expect them to return a specific parameter back, we have to store it locally. This makes the parameter nullable, which is why we need the `UNKNOWN` fallback.

Reply:
Why not make a case of the enum for not being handled and set it as default instead? You are checking for a null parameter, why not checking for the non-handled case?

Reply:
I have a case in `DefaultBrowserPromptsExperimentImpl` to handle the `UNKNOWN` type, if that's what you mean. I can also just make the cache field default to `UNKNOWN` to get rid of the elvis operator, it won't change anything functionally in the end.